proofs/flatten.go

```diff
@@ -135,7 +136,7 @@ func (f *messageFlattener) handleValue(prop Property, value reflect.Value, getSa
 
 		// Append length of slice as tree leaf
 		lengthProp := prop.LengthProp(saltsLengthSuffix)
-		f.appendLeaf(lengthProp, strconv.Itoa(value.Len()), getSalt(lengthProp.CompactName()), saltsLengthSuffix, []byte{}, false)
+		f.appendLeaf(lengthProp, []byte(strconv.Itoa(value.Len())), getSalt(lengthProp.CompactName()), saltsLengthSuffix, []byte{}, false)
```

Comment:
This actually encodes the integer to string and then to bytes. Instead an integer should be converted straight to binary. See: https://stackoverflow.com/questions/16888357/convert-an-integer-to-a-byte-array

(same as below)

Reply:
E.g.: int64(2) should be [0, 0, 0, 0x01] 
NB: Let's use BigEndian encoding.